### PR TITLE
Implement eth_getWork and add route handlers for unsupported methods

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
**Description**:
- Implement `eth_getWork` with a hardcoded return value.
- All non-supported methods return the correct error message.
- Enables custom error messages to be returned, not only the default ones from `koa-jsonrpc`.

**Related issue(s)**:

Fixes #
https://github.com/hashgraph/hedera-json-rpc-relay/issues/55
https://github.com/hashgraph/hedera-json-rpc-relay/issues/56

**Notes for reviewer**:

**Checklist**

- [] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -256,6 +257,57 @@ export class EthImpl implements Eth {
...
+    this.logger.trace('getWork()');
+    return Errors['NO_MINING_WORK'];
+  }
+
+  /**
+   * Unsupported methods always return UNSUPPORTED_METHOD error.
+   */
+  submitHashrate(): JsonRpcError {
+    this.logger.trace('submitHashrate()');
+    return Errors['UNSUPPORTED_METHOD'];
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:273]: Use predefined here and in all the others
```suggestion
    return predefined.UNSUPPORTED_METHOD;
```

File: packages/relay/src/lib/eth.ts [hunk 2]
```diff
@@ -256,6 +257,57 @@ export class EthImpl implements Eth {
     return EthImpl.zeroHex;
   }
 
+  /**
+   * Always returns NO_MINING_WORK error.
+   */
+  getWork(): JsonRpcError {
+    this.logger.trace('getWork()');
+    return Errors['NO_MINING_WORK'];
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:265]:
```suggestion
    return Errors['NO_MINING_WORK'];
```
```suggestion
    return predefined.NO_MINING_WORK
```

File: packages/relay/src/lib/net.ts [hunk 3]
```diff
@@ -38,10 +39,10 @@ export class NetImpl implements Net {
   }
 
   /**
-   * TODO: We should get this value from the address book...
+   * Method is not supported
    */
-  peerCount(): number {
-    return 26;
+  peerCount(): JsonRpcError {
+    return Errors['UNSUPPORTED_METHOD'];
```
Comment (Nana-EC) [inline packages/relay/src/lib/net.ts:45]:
```suggestion
    return predefined.UNSUPPORTED_METHOD;
```

File: packages/relay/src/lib/parity.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+export class ParityImpl implements Parity {
+  private client: Client;
+
+  constructor(client: Client) {
+    this.client = client;
+  }
+
+  nextNonce(): JsonRpcError {
+    return Errors['UNSUPPORTED_METHOD'];
```
Comment (Nana-EC) [inline packages/relay/src/lib/parity.ts:33]:
```suggestion
    return Errors['UNSUPPORTED_METHOD'];
```
```suggestion
    return predefined.UNSUPPORTED_METHOD;
```

File: packages/relay/src/lib/web3.ts [hunk 5]
```diff
@@ -31,4 +32,8 @@ export class Web3Impl implements Web3 {
   clientVersion(): string {
     return 'relay/' + process.env.npm_package_version ?? '';
   }
+
+  sha(): JsonRpcError {
+    return Errors['UNSUPPORTED_METHOD'];
```
Comment (Nana-EC) [inline packages/relay/src/lib/web3.ts:37]:
```suggestion
    return predefined.UNSUPPORTED_METHOD;
```

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looking good.
Some suggestions and missing items

File: packages/relay/src/lib/eth.ts (hunk 2 above)
Comment (shemnon) [reply packages/relay/src/lib/eth.ts:265]: I think this should just be UNSUPPORTED_METHOD

**Comment by shemnon** [comment]
Here's a complete list from execution-apis, If it's not on this list and it's in the `Eth` handler my knee-jerk reaction is we should delete the API.   
```
"eth_getBlockByHash"
"eth_getBlockByNumber"
"eth_getBlockTransactionCountByHash"
"eth_getBlockTransactionCountByNumber"
"eth_getUncleCountByBlockHash"
"eth_getUncleCountByBlockNumber"
"eth_protocolVersion"
"eth_chainId"
"eth_syncing"
"eth_coinbase"
"eth_accounts"
"eth_blockNumber"
"eth_call"
"eth_estimateGas"
"eth_gasPrice"
"eth_feeHistory"
"eth_newFilter"
"eth_newBlockFilter"
"eth_newPendingTransactionFilter"
"eth_uninstallFilter"
"eth_getFilterChanges"
"eth_getFilterLogs"
"eth_getLogs"
"eth_mining"
"eth_hashrate"
"eth_getWork"
"eth_submitWork"
"eth_submitHashrate"
"eth_sign"
"eth_signTransaction"
"eth_getBalance"
"eth_getStorageAt"
"eth_getTransactionCount"
"eth_getCode"
"eth_sendTransaction"
"eth_sendRawTransaction"
"eth_getTransactionByHash"
"eth_getTransactionByBlockHashAndIndex"
"eth_getTransactionByBlockNumberAndIndex"
"eth_getTransactionReceipt"
```

**Comment by Nana-EC** [comment]
> Here's a complete list from execution-apis, If it's not on this list and it's in the `Eth` handler my knee-jerk reaction is we should delete the API.
> 
> ```
> "eth_getBlockByHash"
> "eth_getBlockByNumber"
> "eth_getBlockTransactionCountByHash"
> "eth_getBlockTransactionCountByNumber"
> "eth_getUncleCountByBlockHash"
> "eth_getUncleCountByBlockNumber"
> "eth_protocolVersion"
> "eth_chainId"
> "eth_syncing"
> "eth_coinbase"
> "eth_accounts"
> "eth_blockNumber"
> "eth_call"
> "eth_estimateGas"
> "eth_gasPrice"
> "eth_feeHistory"
> "eth_newFilter"
> "eth_newBlockFilter"
> "eth_newPendingTransactionFilter"
> "eth_uninstallFilter"
> "eth_getFilterChanges"
> "eth_getFilterLogs"
> "eth_getLogs"
> "eth_mining"
> "eth_hashrate"
> "eth_getWork"
> "eth_submitWork"
> "eth_submitHashrate"
> "eth_sign"
> "eth_signTransaction"
> "eth_getBalance"
> "eth_getStorageAt"
> "eth_getTransactionCount"
> "eth_getCode"
> "eth_sendTransaction"
> "eth_sendRawTransaction"
> "eth_getTransactionByHash"
> "eth_getTransactionByBlockHashAndIndex"
> "eth_getTransactionByBlockNumberAndIndex"
> "eth_getTransactionReceipt"
> ```

I'm in agreement.
We can add anything not in the list later when needed

File: packages/relay/src/lib/eth.ts [hunk 6]
```diff
@@ -251,6 +252,47 @@ export class EthImpl implements Eth {
     return EthImpl.zeroHex;
   }
 
+  /**
+   * Always returns NO_MINING_WORK error.
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:256]: nit:
```suggestion
   * Always returns UNSUPPORTED_METHOD error.
```

File: packages/relay/tests/helpers.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import {expect} from "chai";
```
Comment (Nana-EC) [inline packages/relay/tests/helpers.ts:21]: This file won't get picked up by the CLI test run since it's looking for `*.spec.ts`

**Review by Nana-EC — CHANGES_REQUESTED** [review]
LG
Just a quick file extension rename and an updated comment

File: packages/relay/tests/helpers.ts (hunk 7 above)
Comment (Ivo-Yankov) [reply packages/relay/tests/helpers.ts:21]: That file doesn't really have any tests, just a helper function to prevent code duplication.

Comment (Nana-EC) [reply packages/relay/tests/helpers.ts:21]: Ahh yeah correct.

**Pull request merged**